src/ConstructiveSolidGeometry/SurfacePrimitives/SphereMantle.jl

```diff
@@ -54,4 +54,9 @@ function sample(s::SphereMantle{T}, g::CartesianTicksTuple{T})::Vector{Cartesian
         for x in _get_x_at_z(s, g, z)
         for y in _get_y_at_z(s, z, x)
     ]
-end
\ No newline at end of file
+end
+
+function distance_to_surface(point::AbstractCoordinatePoint{T}, s::SphereMantle{T})::T where {T}
+    pcy = CylindricalPoint(point)
+    return abs(hypot(pcy.r, pcy.z) - s.r)
+end
```

Comment:
I would not convert `CartesianPoint` to `CylindricalPoint` here but define a separate function:
```julia
function distance_to_surface(point::CartesianPoint{T}, s::SphereMantle{T})::T where {T}
    return abs(norm(point) - s.r)
end
```